### ISSUE TITLE
Enhancement: Select All Above/Below Playhead

### ISSUE THREAD

**Issue by Average-FNF-Modder** [FunkinCrew/Funkin#2644]
## What is your suggestion, and why should it be implemented?

I suggest a selection option in Edit that selects everything either above or below the current playhead, allowing you to move one whole half of a chart down or something with ease. Would be especially useful when charting erect remixes using the OG charts as skeletons when they have solo parts you'd want to make room for, but I'm sure it has other uses.

**Comment by EliteMasterEric** [comment]
I feel like I chose a keybind for this at one point but never got around to implementing it.

Might be a good thing to add to the mostly-unused context menu that appears when you right click?

**Comment by Hundrec** [comment]
I think it could go better under the Edit menu since shift-right click feels less intuitive for this sort of thing.

**Comment by Lasercar** [comment]
> I feel like I chose a keybind for this at one point but never got around to implementing it.
> 
> Might be a good thing to add to the mostly-unused context menu that appears when you right click?

Might I ask what the keybind is/was? I wanna take a shot at doing this.

**Comment by EliteMasterEric** [comment]
> > I feel like I chose a keybind for this at one point but never got around to implementing it.
> > Might be a good thing to add to the mostly-unused context menu that appears when you right click?
> 
> Might I ask what the keybind is/was? I wanna take a shot at doing this (I don't want to modify the assets and add a button for it yet).
> 
> Hmm, I've only just noticed that there's 3 greyed out buttons in the edit tab, and the select before/after cursor sound like this suggestion, so maybe those were the hotkeys?

Correct, the "Select Before Cursor" and "Select After Cursor" (bound to Shift-Home and Shift-End) respectively are intended to select all the notes before the cursor and all the notes after the cursor respectively, with Ctrl acting as expected.

**Comment by Lasercar** [comment]
> > Hmm, I've only just noticed that there's 3 greyed out buttons in the edit tab, and the select before/after cursor sound like this suggestion, so maybe those were the hotkeys?
> 
> Correct, the "Select Before Cursor" and "Select After Cursor" (bound to Shift-Home and Shift-End) respectively are intended to select all the notes before the playhead and all the notes after the playhead respectively, with Ctrl acting as expected.
> 
> It would also be cool to have a similar function bound to the context menu, where right clicking a note would allow you to select all the notes before/after it.
> 
> Like other selection inputs, it should utilize the Command system so it gets added to the history and can easily be undone or redone.


Ooooh, I was thinking of hard coding it to use the playhead position, but given that you said that you'd also like it to work on notes, I'll make it work with a time that gets passed into it.

Hmm, the deselect version would have to be separate command still, but I'm sure that'll be fine.
